TRACING-3279: Add documentation for Batch processor for OTEL (#64689)
openshift/openshift-docs: modules/distr-tracing-otel-config-collector.adoc

[id="processors_{context}"]
=== Processors

[id="batch-processor_{context}"]
==== Batch processor

> The Resource Detection processor is designed to identify host resource details in alignment with OpenTelemetry's resource semantic standards. Using this detected information, it can add or replace the resource values in telemetry data.

Reviewer: Looks like a copy&paste error?

Reviewer 2: Thank you for spotting this, Andreas.

Author: Sorry. It seems when I moved the PR from one branch to another I messed up

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
* Supported signals: traces, metrics

> .OpenTelemetry Collector using the Batch processor

Reviewer 2: ```suggestion
Example of the .OpenTelemetry Collector custom resource when using the Batch processor
```

> .OpenTelemetry Collector using the Resource Detection processor

Reviewer: ```suggestion
.OpenTelemetry Collector using the Batch processor
```

[source,yaml]
----
  config: |
    processor:
      batch:
        timeout: 5s
        send_batch_max_size: 10000
    service:
      pipelines:
        traces:
          processors: [batch]
        metrics:
          processors: [batch]
----

.Parameters used by the Batch processor
[cols="3",options="header"]

> |============================================================================================================================================================================================================

Reviewer 2: ```suggestion
|===
```

|Parameter |Description |Default
| timeout                    | Send the batch after a specific time duration, irrespective of its size.                                                                                               | 200ms
| send_batch_size            | Send the batch of telemetry data after the desired number of spans or metrics                                                                                          | 8192 
| send_batch_max_size        | The maximum allowable size of the batch. Must be equal or greater than send_batch_size.                                                                                | 0    
| metadata_keys              | When activated, this processor will generate a batcher instance for each unique set of values found in the client.Metadata.                                            | []   

> | metadata_cardinality_limit | When the metadata_keys are populated, this configuration restricts the number of distinct metadata key value combinations processed throughout the process's duration. | 1000 

Author: > BTW, I didn't understand the word batcher

Even the upstream documentation used that term:
https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md

It is like: you have a batch but, if you uses that parameter, you will have multiple "batch queues". I'm totally open to change it but I was not able to find a better wording. Any idea?

> |============================================================================================================================================================================================================

Reviewer 2: ```suggestion
|===
```

[id="processors_{context}"]
=== Processors

[id="batch-processor_{context}"]
==== Batch processor

The batch processor batches the data to reduce the number of outgoing connections needed to transfer the telemetry information.

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
* Supported signals: traces, metrics

> .Example of the OpenTelemetry Collector custom resource when using the Batch processor

Reviewer 2: ```suggestion
.Example of the OpenTelemetry Collector custom resource when using the batch processor
```

> .Example of the OpenTelemetry Collector custom resource when using the Batch processor

Reviewer 2: Looks like it's a lowercase "batch processor":
https://opentelemetry.io/search/?q=batch%20processor
https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor
```suggestion
.Example of the OpenTelemetry Collector custom resource when using the batch processor
```

[source,yaml]
----
  config: |
    processor:
      batch:
        timeout: 5s
        send_batch_max_size: 10000
    service:
      pipelines:
        traces:
          processors: [batch]
        metrics:
          processors: [batch]
----

> .Parameters used by the Batch processor

Reviewer 2: ```suggestion
.Parameters used by the batch processor
```

> .Parameters used by the Batch processor

Reviewer 2: Looks like it's a lowercase "batch processor":
https://opentelemetry.io/search/?q=batch%20processor
https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/batchprocessor
```suggestion
.Parameters used by the batch processor
```

> .Parameters used by the Batch processor

Reviewer 2: Let's put it in lowercase for now (I'll review all these names later):
```suggestion
.Parameters used by the batch processor
```

[cols="3",options="header"]
|===
|Parameter |Description |Default

| `timeout`
| Sends the batch after a specific time duration, irrespective of its size.
| 200ms

| `send_batch_size`
| Sends the batch of telemetry data after the specified number of spans or metrics.
| 8192

| `send_batch_max_size`
| The maximum allowable size of the batch. Must be equal or greater than `send_batch_size`.

> | 0

Reviewer 2: I'm curious--what are the units for this parameter?

Author: Number of spans, metric data points, or log records

[id="processors_{context}"]
=== Processors

[id="batch-processor_{context}"]
==== Batch processor

The batch processor batches the data to reduce the number of outgoing connections needed to transfer the telemetry information.

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
* Supported signals: traces, metrics

.Example of the OpenTelemetry Collector custom resource when using the batch processor
[source,yaml]
----
  config: |
    processor:
      batch:
        timeout: 5s
        send_batch_max_size: 10000
    service:
      pipelines:
        traces:
          processors: [batch]
        metrics:
          processors: [batch]
----

.Parameters used by the batch processor
[cols="3",options="header"]
|===
|Parameter |Description |Default

| `timeout`
| Sends the batch after a specific time duration, irrespective of its size.
| 200ms

| `send_batch_size`
| Sends the batch of telemetry data after the specified number of spans or metrics.
| 8192

| `send_batch_max_size`

> | The maximum allowable size of the batch in terms of the number of spans, metric data points, or log records. Must be equal or greater than `send_batch_size`.

Reviewer 2: ```suggestion
| The maximum allowable size of the batch in terms of the number of spans, metric data points, or log records. Must be equal to or greater than `send_batch_size`.
```

| 0

| `metadata_keys`
| When activated, a batcher instance is created for each unique set of values found in the `client.Metadata`.
| []

| `metadata_cardinality_limit`

> | When the `metadata_keys` are populated, this configuration restricts the number of distinct metadata key-value combinations processed throughout the duration of the process.

Reviewer 2: ```suggestion
| When the `metadata_keys` are populated, this configuration restricts the number of processed distinct metadata key-value combinations during the process.
```
